packages/utils/telemetry-utils/src/test/telemetryEventBatcher.spec.ts

```diff
@@ -56,12 +55,316 @@ describe("TelemetryEventBatcher", () => {
+		assert.strictEqual(loggedEvent.avgpropertyTwo, 15.5);
+		assert.strictEqual(loggedEvent.avgpropertyThree, 105.5);
+		assert.strictEqual(loggedEvent.maxpropertyOne, 10);
+		assert.strictEqual(loggedEvent.maxpropertyTwo, 20);
+		assert.strictEqual(loggedEvent.maxpropertyThree, 110);
+
+		for (let i = 101; i <= threshold + 100; i++) {
+			eventBatcher.measure(() => ({
+				telemetryProperties: {
+					propertyOne: i,
+					propertyTwo: i + 10,
+					propertyThree: i + 100,
+				},
+			}));
+		}
+
+		assert.strictEqual(logger.events.length, 2);
+		loggedEvent = logger.events[1];
+
+		assert.strictEqual(loggedEvent.avgpropertyOne, 105.5);
+		assert.strictEqual(loggedEvent.avgpropertyTwo, 115.5);
+		assert.strictEqual(loggedEvent.avgpropertyThree, 205.5);
+		assert.strictEqual(loggedEvent.maxpropertyOne, 110);
+		assert.strictEqual(loggedEvent.maxpropertyTwo, 120);
+		assert.strictEqual(loggedEvent.maxpropertyThree, 210);
+
+		spy.restore();
+	});
+
+	it("separately emits event in different instances", () => {
+		const thresholdOne = 10;
+		const loggerOne = new TestLogger();
+		const batcherOne = new TelemetryEventBatcher<keyof TestTelemetryProperties>(
+			{ eventName: "testEventOne" },
+			loggerOne,
+			thresholdOne,
+		);
+
+		const thresholdTwo = 20;
+		const loggerTwo = new TestLogger();
+		const batcherTwo = new TelemetryEventBatcher<keyof TestTelemetryProperties>(
+			{ eventName: "testEventTwo" },
+			loggerTwo,
+			thresholdTwo,
+		);
+
+		for (let i = 0; i < thresholdOne - 1; i++) {
+			batcherOne.measure(() => ({
+				eventName: "testCall",
+				telemetryProperties: {
+					propertyOne: 1,
+					propertyTwo: 2,
+					propertyThree: 3,
+				},
+			}));
+		}
+
+		for (let i = 0; i < thresholdTwo - 1; i++) {
+			batcherTwo.measure(() => ({
+				eventName: "testCall",
+				telemetryProperties: {
+					propertyOne: 4,
+					propertyTwo: 5,
+					propertyThree: 6,
+				},
+			}));
+		}
+
+		assert.strictEqual(loggerOne.events.length, 0);
+		assert.strictEqual(loggerTwo.events.length, 0);
+
+		const resultOne = batcherOne.measure(() => ({
+			eventName: "testCall",
+			telemetryProperties: {
+				propertyOne: 1,
+				propertyTwo: 2,
+				propertyThree: 3,
+			},
+		}));
+
+		const resultTwo = batcherTwo.measure(() => ({
+			eventName: "testCall",
+			telemetryProperties: {
+				propertyOne: 4,
+				propertyTwo: 5,
+				propertyThree: 6,
+			},
+		}));
+
+		assert.strictEqual(loggerOne.events.length, 1);
+		assert.strictEqual(loggerTwo.events.length, 1);
+
+		assert.strictEqual(resultOne.telemetryProperties.propertyOne, 1);
+		assert.strictEqual(resultOne.telemetryProperties.propertyTwo, 2);
+		assert.strictEqual(resultOne.telemetryProperties.propertyThree, 3);
+
+		assert.strictEqual(resultTwo.telemetryProperties.propertyOne, 4);
+		assert.strictEqual(resultTwo.telemetryProperties.propertyTwo, 5);
+		assert.strictEqual(resultTwo.telemetryProperties.propertyThree, 6);
+	});
+
+	it("correctly calculates data in different instances", () => {
```

Comment:
I'm not sure this test gives us anything that the other ones don't cover already? We have one that validates correct calculation, and one that validates separate instances log independently. I'd say we can remove this one.

Reply:
Yeah, I wanted to make sure the logger is functioning properly in both situations. But no problem removing it, if it seems redundant.